FSJ_django20_project/FSJ/views_coordinator.py

```diff
@@ -515,3 +518,55 @@ def coordinator_application_list(request, award_idnum):
 
     template = loader.get_template("FSJ/application_list.html")
     return HttpResponse(template.render(context, request))
+
+
+def coordinator_upload_students(request):
+    FSJ_user = get_FSJ_user(request.user.username)
+
+    if request.method == "POST":
+        form = FileUploadForm(request.POST, request.FILES)
+        if form.is_valid():
+
+            try:
+                if 'student_file' in request.FILES:
+                    csv_file = request.FILES['student_file']
+                    csv_file.seek(0)
+                    studentreader = csv.DictReader(io.StringIO(csv_file.read().decode('utf-8-sig')))    
+                    for row in studentreader:
+                        program = Program.objects.get(code = row['Prog'])
+                        year = YearOfStudy.objects.get(year = row['Year'])
+                        obj, created = Student.objects.update_or_create(
+                            ccid = row['CCID'],
+                            defaults={'ualberta_id': row['ID'], 'first_name': row['First Name'], 'last_name': row['Last Name'], 'email' : row['Email (Univ)'],
+                                      'program' : program, 'year' : year,},
+                        ) 
+                        
+                if 'gpa_file' in request.FILES:
+                    csv_file = request.FILES['gpa_file']
+                    csv_file.seek(0)
+                    gpareader = csv.DictReader(io.StringIO(csv_file.read().decode('utf-8-sig')))    
+                    for row in gpareader:
+                        student = Student.objects.get(ccid = row['CCID'])
+                        if row['GPA']:
+                            student.gpa = row['GPA']
+                            student.save()
+                                    
+                return redirect('studentlist')
+            
+            except UnicodeDecodeError:
+                messages.warning(request, 'Please upload a UTF-8 encoded CSV file.')
+                
+            except KeyError:
+                messages.warning(request, 'Please make sure all column names match specified column names.')     
+                
+            except:
+                messages.warning(request, "I don't know how you even got this error.")
```

Comment:
Need to handle case where program or year does not exist adequately, as it is not easy to see when that is an issue.